src/test/ui/consts/issue-55878.stderr

```diff
@@ -0,0 +1,15 @@
+error: reaching this expression at runtime will panic or abort
+  --> $SRC_DIR/libcore/mem/mod.rs:LL:COL
+   |
+LL |     intrinsics::size_of::<T>()
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ rustc layout computation failed: SizeOverflow([u8; 18446744073709551615])
+   | 
+  ::: $DIR/issue-55878.rs:3:26
+   |
+LL |     println!("Size: {}", std::mem::size_of::<[u8; std::u64::MAX as usize]>());
+   |                          ---------------------------------------------------
+   |
+   = note: `#[deny(const_err)]` on by default
```

Comment:
Wait... the issue describes the snippet above as a soundness hole enabling illegal instructions yet this is using a *lint* to ensure soundness?

Reply:
yea, we'd need to change https://github.com/rust-lang/rust/blob/d1d79ae3ad36e82e2be33bea108d6a4e98ddce0b/src/librustc_mir/const_eval.rs#L687, too

but all of this is just papering over the issue. I think you can also ICE the compiler if you do this `size_of` call inside a static.

Note that there is no unsoundness. I'm certain that we are hitting https://github.com/rust-lang/rust/blob/a7f28678bbf4e16893bb6a718e427504167a9494/src/librustc_codegen_ssa/mir/operand.rs#L480-L489

We need to redesign this properly.

@estebank 	what were the tests that broke when you inserted the unconditional `delay_span_bug`?

Reply:
During compilation I got a lot of the following

```
error: internal compiler error: erroneous constant used
   --> src/libcore/mem/mod.rs:243:5
    |
243 |     intrinsics::size_of::<T>()
    |     ^^^^^^^^^^^^^^^^^^^^^^^^

error: internal compiler error: erroneous constant used
   --> src/libcore/num/mod.rs:171:16
    |
171 |         self.0.fmt(f)
    |                ^^^

error: internal compiler error: erroneous constant used
   --> src/libcore/num/dec2flt/algorithm.rs:211:25
    |
211 |                     z = prev_float(z);
    |                         ^^^^^^^^^^
```

Reply:
@Centril turned it into an error

Reply:
> Note that there is no unsoundness. I'm certain that we are hitting 

/me is happy that a block of code commented "Allow RalfJ to sleep soundly" is preventing this from being a soundness issue.  ;)